lgc/builder/DescBuilder.cpp

```diff
@@ -68,7 +68,17 @@ Value *DescBuilder::CreateLoadBufferDesc(unsigned descSet, unsigned binding, Val
   const ResourceNode *node = nullptr;
   if (!m_pipelineState->isUnlinked() || !m_pipelineState->getUserDataNodes().empty()) {
     // We have the user data layout. Find the node.
-    std::tie(topNode, node) = m_pipelineState->findResourceNode(ResourceNodeType::DescriptorBuffer, descSet, binding);
+    ResourceNodeType searchNodeType = ResourceNodeType::DescriptorBuffer;
+    if (flags & BufferFlagConst)
+      searchNodeType = ResourceNodeType::DescriptorConstBuffer;
+    else if (flags & BufferFlagNonConst)
+      searchNodeType = ResourceNodeType::DescriptorBuffer;
+    else if (flags & BufferFlagShaderResource)
+      searchNodeType = ResourceNodeType::DescriptorResource;
+    else if (flags & BufferFlagSampler)
+      searchNodeType = ResourceNodeType::DescriptorSampler;
+
+    std::tie(topNode, node) = m_pipelineState->findResourceNode(searchNodeType, descSet, binding);
```

Comment:
The commit says it's about get.desc.ptr, but here there's a modification on load.buffer.desc. Please separate this out.